src/components/UserOnboarding.js

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useRef } from "react";
+import { useApp } from "../context/app-context";
+
+export const UserOnboarding = () => {
+  const { name, dispatch, time, greet } = useApp();
+  let intervalRef = useRef();
+  useEffect(() => {
+    localStorage.setItem("user", JSON.stringify(name));
+  }, [name]);
+
+  useEffect(() => {
+    intervalRef.current = setInterval(() => {
+      let today = new Date();
+      let hour = today.getHours();
+
+      if (hour >= 12) {
+        dispatch({ type: "SET_GREET", payload: "Good Afternoon" });
+        if (hour > 16) {
+          dispatch({ type: "SET_GREET", payload: "Good Evening" });
+        }
+        // hour = hour - 12;
+        if (hour < 10) {
+          hour = "0" + hour;
+        }
+      }
+
+      if (hour < 12) {
+        dispatch({ type: "SET_GREET", payload: "Good Morning" });
+      }
+
+      let minute = today.getMinutes();
+      if (minute < 10) minute = "0" + minute;
+      dispatch({ type: "SET_TIME", payload: { hour, minute } });
```

Comment:
You can use `toLocaleTimeString` method to get the formmated time
like this `const [hh, mm, ss] = today.toLocaleTimeString().split(":")`

Reply:
sure, I will refactor this part a little, then I will do it.